XiEditor/StyleMap.swift

```diff
@@ -16,11 +16,7 @@ import Cocoa
 
 /// A represents a given text style.
 struct Style {
-    var fgColor: NSColor;
-    var bgColor: NSColor;
-    var weight: UInt16;
-    var underline: Bool;
-    var italic: Bool;
+    var attributes: [String: Any];
```

Comment:
We could also keep all the properties, but instead of creating the attributes array on every call, it is now just created once.